pynpoint/util/multipca.py

```diff
@@ -145,20 +161,34 @@ def run_job(self,
             Output residuals.
         """
 
-        residuals, res_rot = pca_psf_subtraction(images=self.m_star_reshape,
-                                                 angles=self.m_angles,
-                                                 pca_number=int(tmp_task.m_input_data),
-                                                 pca_sklearn=self.m_pca_model,
-                                                 im_shape=self.m_im_shape,
-                                                 indices=self.m_indices)
+        if isinstance(tmp_task.m_input_data, tuple):
+            pca_number = tmp_task.m_input_data
+        else:
+            pca_number = int(tmp_task.m_input_data)
+
+        residuals, res_rot = postprocessor(images=self.m_star_reshape,
+                                           angles=self.m_angles,
+                                           scales=self.m_scales,
+                                           pca_number=pca_number,
+                                           pca_sklearn=self.m_pca_model,
+                                           im_shape=self.m_im_shape,
+                                           indices=self.m_indices,
+                                           processing_type=self.m_processing_type)
+
+        if self.m_processing_type == 'ADI' and res_rot.ndim == 3:
```

Comment:
Add a comment to explain this if conditions

Reply:
Added comment and simplified the expression.